### PR TITLE
[FLINK-34883] Fix postgres uuid column as PK

### PR DESCRIPTION
**Pull request by eskabetxe**
A majority of Postgres databases use UUIDs as primary keys.

When we enable 'scan.incremental.snapshot.enabled = true', Flink-CDC will try to split into chunks.
The splitTableIntoChunks function relies on the queryMinMax function, which fails when trying to calculate the MIN(UUID) and MAX(UUID), as that is not supported in Postgres.

This change those queries to cast the uuid to a String value.

What was done:
- added uuid type for postgres mapped as string
- for max and min queries, the uuid is casted to text: MIN(uuid::text)
- for queries with >= and < we cast the param passed as: uuid >= (?)::uuid 
-- this way we can still use the index and only cast one value by query
- changes on other databases are because we need to change an Splitter interface, to pass the Column instead of columnName

### CODE REVIEW

**Comment by eskabetxe** [comment]
@GOODBOY008 can you help here to find someone that can help review this?

**Comment by eskabetxe** [comment]
@leonardBang can you help here?

**Comment by leonardBang** [comment]
Thanks @eskabetxe for the contribution, I'd like to find someone  who is more familiar to Pg to review this PR.

**Comment by leonardBang** [comment]
@loserwang1024 would you like to review this PR?

**Review by GOODBOY008 — CHANGES_REQUESTED** [review]
@eskabetxe Thanks for your contributions.Sorry for late review, I left some comment.

File: flink-cdc-connect/flink-cdc-source-connectors/flink-connector-postgres-cdc/src/main/java/org/apache/flink/cdc/connectors/postgres/source/fetch/PostgresScanFetchTask.java [hunk 1]
```diff
@@ -280,12 +282,18 @@ private void createDataEventsForTable(
...
+                            .collect(Collectors.toList());
+
             final String selectSql =
                     PostgresQueryUtils.buildSplitScanQuery(
                             snapshotSplit.getTableId(),
                             snapshotSplit.getSplitKeyType(),
                             snapshotSplit.getSplitStart() == null,
-                            snapshotSplit.getSplitEnd() == null);
+                            snapshotSplit.getSplitEnd() == null,
+                            uuidFields);
```
[the remark this answers is not in the document]
Comment (eskabetxe) [reply flink-cdc-connect/flink-cdc-source-connectors/flink-connector-postgres-cdc/src/main/java/org/apache/flink/cdc/connectors/postgres/source/fetch/PostgresScanFetchTask.java:296]: I will need to pass the Table field, and that will also force me to change `PostgresChunkSplitter.buildSplitScanQuery` or create a new method specific this logic.

Current I'm overloading the method to only have to change here..
image

File: flink-cdc-connect/flink-cdc-source-connectors/flink-cdc-base/src/main/java/org/apache/flink/cdc/connectors/base/source/assigner/splitter/JdbcSourceChunkSplitter.java [hunk 2]
```diff
@@ -47,11 +47,10 @@ public interface JdbcSourceChunkSplitter extends ChunkSplitter {
      *
      * @param jdbc JDBC connection.
      * @param tableId table identity.
-     * @param columnName column name.
+     * @param column column name.
```
[the remark this answers is not in the document]
Comment (eskabetxe) [reply flink-cdc-connect/flink-cdc-source-connectors/flink-cdc-base/src/main/java/org/apache/flink/cdc/connectors/base/source/assigner/splitter/JdbcSourceChunkSplitter.java]: done

File: flink-cdc-connect/flink-cdc-source-connectors/flink-cdc-base/src/main/java/org/apache/flink/cdc/connectors/base/source/assigner/splitter/JdbcSourceChunkSplitter.java [hunk 3]
```diff
@@ -60,12 +59,11 @@ Object[] queryMinMax(JdbcConnection jdbc, TableId tableId, String columnName)
...
```
[the remark this answers is not in the document]
Comment (eskabetxe) [reply flink-cdc-connect/flink-cdc-source-connectors/flink-cdc-base/src/main/java/org/apache/flink/cdc/connectors/base/source/assigner/splitter/JdbcSourceChunkSplitter.java]: done

File: flink-cdc-connect/flink-cdc-source-connectors/flink-cdc-base/src/main/java/org/apache/flink/cdc/connectors/base/source/assigner/splitter/JdbcSourceChunkSplitter.java [hunk 4]
```diff
@@ -75,15 +73,15 @@ Object queryMin(
...
```
[the remark this answers is not in the document]
Comment (eskabetxe) [reply flink-cdc-connect/flink-cdc-source-connectors/flink-cdc-base/src/main/java/org/apache/flink/cdc/connectors/base/source/assigner/splitter/JdbcSourceChunkSplitter.java]: done

**Review by GOODBOY008 — APPROVED** [review]

**Comment by eskabetxe** [comment]
@GOODBOY008 any chance to get this merged?

**Pull request merged**